openmrs-watcher/src/main/java/org/openmrs/eip/mysql/watcher/MySqlWatcherEndpointConfigurer.java

```diff
@@ -21,10 +18,8 @@ public boolean configure(CamelContext camelContext, Object target, String name,
 	}
 	
 	@Override
-	public Map<String, Object> getAllOptions(Object target) {
-		Map<String, Object> options = new CaseInsensitiveMap();
-		
-		return options;
+	public Class<?> getOptionType(String name, boolean ignoreCase) {
+		return null;
```

Comment:
Is this the safest override? I'm not aware of the full implication of this, but I'm quite suspicious that returning `null` could bite us in some way.

Reply:
This comes from implementing `PropertyConfigurerGetter`. It has zero usages in the Codebase. Probably I should just throw `UnsupportedOperationException` for now.